lib/generators/react_on_rails/base_generator.rb

```diff
@@ -62,7 +62,7 @@ def update_git_ignore
           /app/assets/javascripts/generated/*
         DATA
 
-        dest_file_exists?(".gitignore") ? append_to_file(".gitignore", data) : puts_setup_file_error(".gitignore", data)
+        dest_file_exists?(".gitignore") ? append_to_file(".gitignore", data) : options.errors.push(return_setup_file_error(".gitignore", data))
```

Comment:
I agree with @justin808, we could create our own class. Either singleton or use one global object should be fine.